catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -367,21 +455,128 @@ function BucketFields({ name, reindex }: BucketFieldsProps) {
         </M.AccordionSummary>
         <M.Box className={classes.group} pt={1}>
           {!!reindex && (
-            <M.Button variant="outlined" fullWidth onClick={reindex}>
-              Re-index and repair
-            </M.Button>
+            <M.Box pb={2.5}>
+              <M.Button variant="outlined" fullWidth onClick={reindex}>
+                Re-index and repair
+              </M.Button>
+            </M.Box>
           )}
+
           <RF.Field
-            component={Form.Field}
-            name="fileExtensionsToIndex"
-            label="File extensions to index (comma-separated)"
-            fullWidth
-            margin="normal"
+            component={Form.Checkbox}
+            type="checkbox"
+            name="enableDeepIndexing"
+            label={
+              <>
+                Enable deep indexing
+                <Hint>Deep indexing help text TBD @akarve pls halp!</Hint>
+              </>
+            }
           />
+
+          <RF.FormSpy subscription={{ modified: true, values: true }}>
+            {({ modified, values }) => {
+              // don't need this while adding a bucket
+              if (!bucket) return null
+              if (
+                !modified?.enableDeepIndexing &&
+                !modified?.fileExtensionsToIndex &&
+                !modified?.indexContentBytes
+              )
+                return null
+              try {
+                if (
+                  R.equals(
+                    bucket.fileExtensionsToIndex,
+                    editFormSpec.fileExtensionsToIndex(values),
+                  ) &&
+                  R.equals(
+                    bucket.indexContentBytes,
+                    editFormSpec.indexContentBytes(values),
+                  )
+                )
+                  return null
+              } catch {
+                return null
+              }
+
+              return (
+                <Lab.Alert
+                  className={classes.warning}
+                  icon={
+                    <M.Icon fontSize="inherit" className={classes.warningIcon}>
+                      error
+                    </M.Icon>
+                  }
+                  severity="warning"
+                >
+                  New deep indexing settings will only affect newly indexed files. You can
```

Comment:
Changing these settings affects files that are indexed after the change. If you wish to deep index existing feels, click "Re-index and repair."